pkgs/development/mobile/gomobile/default.nix

```diff
@@ -0,0 +1,65 @@
+{ stdenv, lib, fetchgit, buildGoPackage, makeWrapper, xcodeenv, androidenv }:
+
+{ xcodeWrapperArgs ? { }
+, xcodeWrapper ? xcodeenv.composeXcodeWrapper xcodeWrapperArgs
+, androidPkgs ? androidenv.androidPkgs_9_0.androidsdk }:
+
+let
+  inherit (stdenv) isDarwin;
+  inherit (lib) optional strings;
+in buildGoPackage rec {
```

Comment:
Can we use buildGoModule?

Reply:
Sure, but I'd like to understand what's the difference. Why would `buildGoModule` be better?

Reply:
I trued out `buildGoModule` but it seems to fail on tests. I'll have to investigate that.
```
=== RUN   TestGobind
=== RUN   TestGobind/GOPATH
    gobind_test.go:117: go tool not available: 'go env GOROOT' does not match runtime.GOROOT:
                go env: /nix/store/pc8lwdfhcljkmmmjnwdanfhp7bqj4lh3-go-1.15.10/share/go
                GOROOT: go
=== RUN   TestGobind/Modules
    gobind_test.go:117: go tool not available: 'go env GOROOT' does not match runtime.GOROOT:
                go env: /nix/store/pc8lwdfhcljkmmmjnwdanfhp7bqj4lh3-go-1.15.10/share/go
                GOROOT: go
--- FAIL: TestGobind (0.10s)
    --- FAIL: TestGobind/GOPATH (0.09s)
    --- FAIL: TestGobind/Modules (0.01s)
=== RUN   TestDocs
=== RUN   TestDocs/GOPATH
    gobind_test.go:151: go tool not available: 'go env GOROOT' does not match runtime.GOROOT:
                go env: /nix/store/pc8lwdfhcljkmmmjnwdanfhp7bqj4lh3-go-1.15.10/share/go
                GOROOT: go
=== RUN   TestDocs/Modules
    gobind_test.go:151: go tool not available: 'go env GOROOT' does not match runtime.GOROOT:
                go env: /nix/store/pc8lwdfhcljkmmmjnwdanfhp7bqj4lh3-go-1.15.10/share/go
                GOROOT: go
--- FAIL: TestDocs (0.10s)
    --- FAIL: TestDocs/GOPATH (0.09s)
    --- FAIL: TestDocs/Modules (0.01s)
FAIL
FAIL    golang.org/x/mobile/cmd/gobind  0.589s
FAIL
```
What's the advantage of `buildGoModule` over `buildGoPackage`?

Reply:
> Why would buildGoModule be better?

It generates a vendor tarball instead of a lockfile.

> I can build it with doCheck = false but not sure if that's advised...

Thats fine if the tests don't work. I am not sure if buildGoPackage exectues tests by default.

Reply:
Okay, i changed it to use `buildGoModule`.

>It generates a vendor tarball instead of a lockfile.

I see, so that's better because it makes updates easier since it doesn't require use of `vgo2nix` correct?
And I guess it also makes the `/nix/store` footprint smaller?